spec/models/program_session_spec.rb

```diff
@@ -1,5 +1,152 @@
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.track_id).to eq(proposal.track_id)
+    end
+
+    it "creates a program session that has the proposal id" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.proposal_id).to eq(proposal.id)
+    end
+
+    it "creates a program session that is active" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.state).to eq("active")
+    end
+
+    it "sets program session id for all speakers" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      session.speakers.each do |speaker|
+        expect(speaker.program_session_id).to eq(session.id)
+        expect(speaker.changed?).to be(false)
+      end
+    end
+
+    it "sets speaker_name from user on each speaker" do
+      proposal.speakers.each do |speaker|
+        expect(speaker.speaker_name).to eq(nil)
+      end
+
+      session = ProgramSession.create_from_proposal(proposal)
+
+      session.speakers.each do |speaker|
+        expect(speaker.speaker_name).to eq(speaker.user.name)
+        expect(speaker.changed?).to be(false)
+      end
+    end
+
+    it "sets speaker_email from user on each speaker" do
+      proposal.speakers.each do |speaker|
+        expect(speaker.speaker_email).to eq(nil)
+      end
+
+      session = ProgramSession.create_from_proposal(proposal)
+
+      session.speakers.each do |speaker|
+        expect(speaker.speaker_email).to eq(speaker.user.email)
+        expect(speaker.changed?).to be(false)
+      end
+    end
+
+    it "does not overwrite name and email if they already exists" do
+      my_proposal = create(:proposal)
+      user = create(:user)
+      create(:speaker,
+              user_id: user.id,
+              event_id: my_proposal.event_id,
+              proposal_id: my_proposal.id,
+              speaker_name: "Fluffy",
+              speaker_email: "fluffy@email.com")
+
+      ProgramSession.create_from_proposal(my_proposal)
+      ps_speaker = my_proposal.speakers.first
+
+      expect(ps_speaker.speaker_name).to eq("Fluffy")
+      expect(ps_speaker.speaker_email).to eq("fluffy@email.com")
+      expect(ps_speaker.changed?).to be(false)
+    end
+
+    it "can handle name but not email already being set" do
+      my_proposal = create(:proposal)
+      user = create(:user, name: "Fluffy", email: "fluffy@email.com" )
+      create(:speaker,
+              user_id: user.id,
+              event_id: my_proposal.event_id,
+              proposal_id: my_proposal.id,
+              speaker_name: "Unicorn")
+
+      ProgramSession.create_from_proposal(my_proposal)
+      ps_speaker = my_proposal.speakers.first
+
+      expect(ps_speaker.speaker_name).to eq("Unicorn")
+      expect(ps_speaker.speaker_email).to eq("fluffy@email.com")
+      expect(ps_speaker.changed?).to be(false)
+    end
+
+    it "can handle email but not name already being set" do
```

Comment:
These last couple tests seem to cover some similar ground. Try breaking the first one (the one that tests name and email being already set) into two separate tests. One for name and one for email. Then the next two can probably be dropped.